Addresses issue #4362

GonzoDMX opened the pull request:
#### What does this PR do and why is it necessary?
Fixes the normalize_version function in src/util/version.py so that it returns the correct version when the input version string is a release candidate containing the characters "-rc".

#### How was it tested? How can it be tested by the reviewer?
Tested in OctoPrint dev environment.

#### What are the relevant tickets if any?
Fixes #4362

cp2004 commented on src/octoprint/util/version.py:
```diff
@@ -208,6 +208,9 @@ def is_prerelease(version_string):
 
 
 def normalize_version(version):
+    if "-rc" in version:
+        version = version.replace("-rc", "rc")
+
```
This only covers the very specific case of having `-rc`.

By PEP440 version spec, https://www.python.org/dev/peps/pep-0440/#pre-release-separators, there could be `-a`, `-b`, `-c`, `-alpha`, `-beta`, `-pre`, `-preview`, `-rc` and possibly more I've missed off.

Probably better to either leave the `-` in (it should be parsed properly by `pkg_resources` or whatever's doing it) or always remove it.

GonzoDMX replied:
I agree this is very naive attempt at fixing the issue, initially I had proposed a more general solution along those lines however, leaving the `-` will fail the unit test and simply removing the `-` for all cases will fail the unit test. I changed to commit to see if this would pass.
I believe leaving it in or simply removing it for all cases should work fine but it will fail a unit test... so maybe that's what needs to change?

GonzoDMX replied:
So more details on the unitest, it tests the string "1.6.0-dev123" and expects "1.6.0". Problem is I believe `-dev` is valid in PEP440.
So the best solution is probably to correct the test and leave in the `-`

GonzoDMX commented:
I've updated the PR, upon further assessment of the problem I believe removing the `-` char is redundant. The version is passed immediately after to pkg_resources.parse_version() which does this for us already.
I've updated the unit test for the function normalize_versions() to remove the case testing the `-` character.

cp2004 reviewed (approved):
I think that's the right way to go 👍
